sound/soc/sof/ipc4-loader.c

```diff
@@ -383,6 +383,11 @@ int sof_ipc4_query_fw_configuration(struct snd_sof_dev *sdev)
 			if (!ipc4_data->max_libs_count)
 				ipc4_data->max_libs_count = 1;
 			break;
+		case SOF_IPC4_FW_CFG_MAX_PPL_COUNT:
+			ipc4_data->max_num_pipelines = *tuple->value;
+			trace_sof_ipc4_fw_config(sdev, "Max PPL count %d",
+						 ipc4_data->max_num_pipelines);
```

Comment:
> check max_num_pipelines to avoid unexpected behavior.

Thanks @RanderWang I added a test before
```
		swidget->instance_id = ida_alloc_max(&ppl_ida, ipc4_data->max_num_pipelines,
						     GFP_KERNEL);
```

Reply:
I would have verified it here and fail early. You can not allocate anyways if `ipc4_data->max_num_pipelines <= 0`